src/powershell/Program.cs

```diff
@@ -19,7 +20,236 @@ public sealed class ManagedPSEntry
         /// </param>
         public static int Main(string[] args)
         {
+            /*
+            Console.WriteLine($"PID: {System.Diagnostics.Process.GetCurrentProcess().Id}");
+            while (!System.Diagnostics.Debugger.IsAttached)
+            {
+                System.Threading.Thread.Sleep(500);
+            }
+            */
+#if UNIX
+            if (HasLoginSpecified(args, out int loginIndex))
+            {
+                return ExecPwshLogin(args, loginIndex);
+            }
+#endif
             return UnmanagedPSEntry.Start(string.Empty, args, args.Length);
         }
+
+#if UNIX
+        /// <summary>
+        /// Checks args to see if -Login has been specified.
+        /// </summary>
+        /// <param name="args">Arguments passed to the program.</param>
+        /// <param name="loginIndex">The arg index (in argv) where -Login was found.</param>
+        /// <returns></returns>
+        private static bool HasLoginSpecified(string[] args, out int loginIndex)
+        {
+            loginIndex = -1;
+
+            // Parameter comparison strings, stackalloc'd for performance
+            for (int i = 0; i < args.Length; i++)
+            {
+                string arg = args[i];
+
+                // Must look like '-<name>'
+                if (arg == null || arg.Length < 2 || arg[0] != '-')
+                { 
+                    continue;
+                }
+
+                // Check for "-Login" or some prefix thereof
+                if (IsParam(arg, "login", "LOGIN"))
```

Comment:
Two reasons:
- `MatchSwitch()` lives in another assembly, and loading it will slow startup
- `MatchSwitch()` allocates and is more general code than this method, which is tailored to the three parameters we care about. I actually am still thinking about inlining it as direct array indexing